Support multi_camera and depth_array render mode for dm environments

realquantumcookie opened the pull request:
# Description

This pull request implements #67 

## Type of change

- New feature (non-breaking change which adds functionality)

### Screenshots

Before: Only one camera supported in `DmControlCompatibilityV0`
![Before](https://user-images.githubusercontent.com/18230652/232661503-c5ddca92-65d5-4fbd-8589-548d63caf28f.png)

After: 

Stacking cameras in `DmControlCompatibilityV0` by using `multi_camera` render mode
![After](https://user-images.githubusercontent.com/18230652/232661627-03e7c1e1-33de-4b91-9ebe-fb80401128cf.png)

Also, Depth Image Rendering in `DmControlCompatibilityV0` by using `depth_array` render_mode

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have run `pytest -v` and no errors are present.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I solved any possible warnings that `pytest -v` has generated that are related to my code to the best of my knowledge.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

pseudo-rnd-thoughts reviewed (changes requested):
Thanks for the changes looks good.
Could you add a quick test to check that this code works, i.e., outputs the expected type and size with several different, `ncam` values

elliottower commented:
Thanks for the contribution, can you run pre-commit hooks? See the readme in docs if you need help installing @realquantumcookie

realquantumcookie commented:
Hi guys, I changed a few things, so
1. I added in two test cases covering rendering of "multi_camera" and "depth_array" mode
    - However it looks like adding new cameras and testing them in “multi_camera" is a bit of a hassle - so are those two tests good enough to cover those two modes?
2. I ran pre-commit hooks.

pseudo-rnd-thoughts reviewed (commented):
I just realised, could you add documentation to the class docstring explaining the different render modes then I think we are good to merge

pseudo-rnd-thoughts reviewed (approved):
@elliottower Has volunteered to do the documentation

elliottower commented:
Btw @realquantumcookie if you have any interest in contributing to Shimmy or other Farama projects in the future feel free to join the discord: https://discord.gg/nhvKkYa6qX (I was also going to shoot you a random personal message about your site/github readme cause I thought they looked nice, but couldn't find any contact info/method)
